scripts/bin/update-packages

```diff
@@ -176,23 +287,30 @@ _fetch_and_cache_tags() {
 		termux_pkg_auto_update
 	}
 
+	# This function generates a ninja file for all packages in the ${__PACKAGES[@]}
+	# So shut up shellcheck nagging us about using `sed`
+	# shellcheck disable=SC2001
 	__generate__() {
 		echo "rule update"
-		echo " command = bash -c \"\$\$F\" -- \$pkg_dir ||:"
+		echo " command = bash -c \"\$\$F\" -- \$pkg_dir || :"
 		sed 's/[^ ]\+/\nbuild &: update\n pkg_dir=&/g' <<< "${__PACKAGES[@]}"
-		echo "build run_all: phony ${__PACKAGES[@]}"
+		echo "build run_all: phony ${__PACKAGES[*]}"
 		echo "default run_all"
 	}
-	
-	local LATEST_TAGS="$(\
+
+	local LATEST_TAGS LATEST_TAGS_GITHUB
+	LATEST_TAGS="$(
 		F="$(declare -p TERMUX_SCRIPTDIR GITHUB_TOKEN TERMUX_REPOLOGY_DATA_FILE); $(declare -f __main__ | sed 's/__main__ ()//')" \
-		env --chdir=/tmp  ninja -f /dev/stdin <<< "$(__generate__)" |& grep "^PKG|"
+		env --chdir=/tmp ninja -f /dev/stdin < <(__generate__) |& grep "^PKG|"
 	)"
-	
-	unset -f __main__  __generate__
+	unset -f __main__ __generate__
+
+	LATEST_TAGS_GITHUB="$(:)" # TODO: abstract the bullshit above into a new script/function
```

Comment:
I wrote this code to avoid storing 2.5k+ lines in a variable. It happened before I realized all package building logs are stored in variables so it is not a problem now.
We can rewrite this code to generate ninja script without the nested function bullshit.

Reply:
We should probably have a look at this `ninja` build monstrosity.
But for the time being, it does its job and isn't what we're focussing on in this PR.

Reply:
In the case if we do [this](https://github.com/termux/termux-packages/pull/22994#discussion_r1960149664) we will not need this code at all.